src/bases/base.js

```diff
@@ -1,20 +1,8 @@
 import basex from '../../vendor/base-x.js'
 import { coerce } from '../bytes.js'
-
-/**
- * @typedef {import('./interface').BaseEncoder} BaseEncoder
- * @typedef {import('./interface').BaseDecoder} BaseDecoder
- * @typedef {import('./interface').BaseCodec} BaseCodec
- */
-
-/**
- * @template {string} T
- * @typedef {import('./interface').Multibase<T>} Multibase
- */
-/**
- * @template {string} T
- * @typedef {import('./interface').MultibaseEncoder<T>} MultibaseEncoder
- */
+// Linter can see that API is used in types.
+// eslint-disable-next-line
+import * as API from '../interface.js'
```

Comment:
In this instance, aren't all of the `API.*` uses just using types from `./interface.ts`? So we now have monolithic type suite that pulls in all types and then have to consume that all over the place? I know for bundlers this is not a concern because `interface.js` is empty, but even for doing localised type checks it seems unnecessary to be loading all types just to test types local to this directory. No? Or are we trying to avoid an `interface.js` alongside every `interface.ts`?

Pros and cons on this decision would be nice to hear.